src/pages/index.js

```diff
@@ -5,28 +5,78 @@ import Section from "../components/Section.js";
 import {
   initialCards,
   selectors,
-  addCardImgUrlInput,
-  addCardTitleInput,
+  imageProfileAvatar,
   nameInput,
   jobInput,
   ImagePopupSelector,
   addCardButton,
   config,
   addCardFormElement,
+  editAvatarFormElement,
   editProfileFormElement,
   editProfileButton,
 } from "../utils/Constants.js";
 import Card from "../components/Card.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
+import { Api } from "../components/Api.js";
+import DeleteConfirmationPopup from "../components/DeleteConfirmationPopup.js";
 
 const handleImageClick = (data) => {
   imagePopup.open(data);
 };
 
 const createCard = (cardData) => {
-  const card = new Card(cardData, selectors.Cardtemplate, handleImageClick);
+  const card = new Card(
+    cardData,
+    selectors.Cardtemplate,
+    handleImageClick,
+    function handleDeleteModal() {
+      deleteCardForm.open();
+
+      deleteCardForm.setSubmitAction(() => {
+        deleteCardForm.renderLoading(true, "Deleting...");
+        api
+          .deleteCard(card.id)
+          .then((res) => {
+            console.log("The delete card function is being called");
+            // delete the card element
+            card.deleteCard();
+          })
+          .then(() => {
+            deleteCardForm.renderLoading(false);
```

Comment:
`.renderLoading(false);` should be in block `finally`


You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

Reply:
Closing should be in `then` - so, everything is correct with `.close()`. Don't move it

Reply:
Done/